### PR TITLE
feat(ns-workflows-1): Add Criterion Object

### PR DESCRIPTION
**Pull request by frankkilcommins**
Refs: #3392

Add Criterion Object to apidom-ns-workflows-1

### CODE REVIEW

File: packages/apidom-ns-workflows-1/src/predicates.ts [hunk 1]
```diff
@@ -60,3 +61,14 @@ export const isSourceDescriptionsElement = createPredicate(
...
 );
+
+export const isCriterionElement = createPredicate(
+  ({ hasBasicElementProps, isElementType, primitiveEq, hasClass }) => {
+    return (element: unknown): element is CriterionElement =>
+      element instanceof CriterionElement ||
+      (hasBasicElementProps(element) &&
+        isElementType('criterion', element) &&
+        primitiveEq('object', element) &&
+        hasClass('criterion', element));
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/src/predicates.ts:72]: There is no class: `criterion` attached to `CriterionElement`.

File: packages/apidom-ns-workflows-1/test/predicates.ts [hunk 2]
```diff
@@ -287,4 +289,62 @@ describe('predicates', function () {
...
+        assert.isFalse(isCriterionElement([]));
+        assert.isFalse(isCriterionElement('string'));
+      });
+    });
+
+    specify('should support duck-typing', function () {
+      const CriterionElementDuck = {
+        _storedElement: 'criterion',
+        _content: [],
+        classes: new ArrayElement(['criterion']),
```
Comment (char0n) [inline packages/apidom-ns-workflows-1/test/predicates.ts:326]: Not in original `CriterionElement` definition.

**Review by char0n — CHANGES_REQUESTED** [review]
LGTM. There is one issue in the predicate and we didn't check the chebox for Criterion Object in the README (tracking progress)...

File: packages/apidom-ns-workflows-1/src/predicates.ts (hunk 1 above)
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/src/predicates.ts:72]: added class

File: packages/apidom-ns-workflows-1/test/predicates.ts (hunk 2 above)
Comment (frankkilcommins) [reply packages/apidom-ns-workflows-1/test/predicates.ts:326]: added to definition

**Review by char0n — APPROVED** [review]
LGTM

**Pull request merged**